packages/core/src/app.ts

```diff
@@ -253,8 +253,8 @@ export class AppBuilder<TreeNames> {
         relativePath: '_testing_suffix_.js',
         source: `
         var runningTests=true;
-        if (window.Testem) {
-          window.Testem.hookIntoTestFramework();
+        if (typeof Testem !== 'undefined' && (typeof QUnit !== 'undefined' || typeof Mocha !== 'undefined')) {
+          Testem.hookIntoTestFramework();
         }`,
```

Comment:
FWIW, I think this is transitionary. Once I can roll out ember-qunit@5.0.0 more broadly, we can remove this completely (and release a breaking change release).